packages/relay-sdk/src/index.ts

```diff
@@ -0,0 +1,132 @@
+
+import { RELAY_URL_HABANERO, RELAY_URL_MANZANO } from '@lit-protocol/constants';
+import { assertNotNull, assertStatus, assertString, getProp } from './util';
+
+type SupportedNetworks = 'habanero' | 'manzano';
+
+function getRelayURLByNetwork(network: SupportedNetworks): string {
+    switch (network) {
+        case 'habanero':
+            return RELAY_URL_HABANERO;
+        case 'manzano':
+            return RELAY_URL_MANZANO;
+    }
+}
+
+export class LitRelayClient {
+
+    private readonly baseUrl: string;
+    private readonly apiKey: string;
+
+    private payerSecret: string | undefined = undefined;
+
+    /**
+     * Create a new LitRelayClient instance. Requires that the payer is already registered.
+     * and the the payer secret is known.
+     * 
+     * ```typescript
+     * const client = new LitRelayClient('https://habanero.lit.dev', 'my-api-key', 'my-payer-secret');
+     * ```
+     * 
+     * @param baseUrl
+     * @param apiKey
+     * @param payerSecret
+     */
+    private constructor(baseUrl: string, apiKey: string, payerSecret: string) {
+        this.baseUrl = baseUrl;
+        this.apiKey = apiKey;
+
+        if (payerSecret) {
+            this.payerSecret = payerSecret;
+        }
+    }
+
+    /**
+     * Adds a new payee to the payer's delegation list.
+     * 
+     * ```typescript
+     * const client = await LitRelayClient.connect('habanero', 'my-api-key', 'my-payer-secret');
+     * const result = await client.addPayee('payee-wallet-address');
+     * ```
+     * @param payeeAddress
+     * 
+     * @returns Promise<{ tokenId: string } | Error>
+     */
+    public addPayee(payeeAddress: string): Promise<{ tokenId: string }> {
+        if (!this.payerSecret) {
+            return Promise.reject('Payer secret key is missing');
+        }
+
+        return fetch(`${this.baseUrl}/add-users`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'api-key': this.apiKey,
+                'payer-secret-key': this.payerSecret,
+            },
+            body: JSON.stringify([payeeAddress]),
+        })
+            .then(assertStatus(200, 'Failed to add payee: request failed'))
+            .then(res => res.json())
+            .then(json => assertString(json.tokenId, 'Failed to add payee: missing token ID'))
+            .then(tokenId => ({ tokenId }));
+    }
+
+    /**
+     * Registers a new payer with the Lit Relay server using the provided API key. Returns
+     * a new LitRelayClient instance with the payer secret key.
+     * 
+     * ```typescript
+     * const client = await LitRelayClient.register('habanero', 'my-api-key');
+     * ```
+     * 
+     * @param network 
+     * @param apiKey 
+     * 
+     * @returns LitRelayClient
+     */
+    public static register(network: SupportedNetworks, apiKey: string): Promise<LitRelayClient {
```

Comment:
A couple things here:

1. Rather than imperatively return a settled promise that has been resolved or rejected, this function should just be an `async` function and use `return` or  `throw` as necessary.  Otherwise we risk the function not returning a Promise if something unexpected happens.  For example, the way this is written, if `getRelayUrlByNetwork()` was to `throw` for any reason, this function would actually throw an error instead of returning a rejected Promise, and that would mean that anyone using Promise chaining methods on the return value of the function would get an error 'could not call .catch'; or 'could not call then' on {} (whatever was thrown).  I recognize that `getRelayUrlByNetwork()` won't currently throw, but we want to make our code robust and it would surprise someone that `throw`'ing from that method would make `register()` no longer return a promise -- an easy avenue for bugs to creep into the codebase when things change later on.
2. Throwing a `Promise.reject()` with a string argument  is the equivalent of `throw`'ing a string, which is an antipattern -- we _always_ want to throw Error objects, so we have a stack trace when we `catch()` it later.  By making this function `async`, you don't need to use `Promise.reject()` or `Promise.resolve()` anywhere in the function body; all sync `return`s and `throw`'s are converted into a Promise with the appropriate state.
3. Is this function signature missing a `>` after `LitRelayClient`? Curious that this would even compile 🤣